user-interface/src/main/java/life/qbic/datamanager/views/projects/project/access/ProjectAccessComponent.java

```diff
@@ -89,91 +92,137 @@ protected ProjectAccessComponent(
     Span titleField = new Span();
     titleField.setText("Project Access Management");
     titleField.addClassName("title");
-
     buttonBar = new Span();
+    Button addCollaboratorButton = new Button("Add people");
+    addCollaboratorButton.addClickListener(event -> openAddCollaboratorDialog());
+    buttonBar.add(addCollaboratorButton);
     header.add(titleField, buttonBar);
     add(header);
     Span userProjectAccessDescription = new Span("Users with access to this project");
-
-    projectCollaborators = projectCollaboratorGrid();
-    add(userProjectAccessDescription, projectCollaborators);
+    projectUserGrid = new Grid<>(ProjectUser.class);
+    initProjectUserGrid();
+    add(userProjectAccessDescription, projectUserGrid);
   }
 
-  private boolean isCurrentUser(ProjectAccessService.ProjectCollaborator collaborator) {
-    var userId = this.authenticationToUserIdTranslator.translateToUserId(
-        SecurityContextHolder.getContext().getAuthentication()).orElseThrow();
-    return Objects.equals(collaborator.userId(), userId);
+  public void setContext(Context context) {
+    if (context.projectId().isEmpty()) {
+      throw new ApplicationException("no project id in context " + context);
+    }
+    this.context = context;
+    setProjectInformation();
   }
 
-  private Button addCollaboratorButton() {
-    Button button = new Button("Add people");
-    button.addClickListener(event -> openAddCollaboratorDialog());
-    return button;
+  private void setProjectInformation() {
+    refreshProjectUserGrid();
+    showControls(userPermissions.changeProjectAccess(context.projectId().orElseThrow()));
   }
 
-  private void showControls() {
-    buttonBar.removeAll();
-    buttonBar.add(addCollaboratorButton());
+  private void refreshProjectUserGrid() {
+    loadProjectUsers();
+    projectUserGrid.setItems(loadProjectUsers());
   }
 
-  private void removeControls() {
-    buttonBar.removeAll();
+  private boolean isCurrentUser(ProjectUser projectUser) {
+    var userId = this.authenticationToUserIdTranslator.translateToUserId(
+        SecurityContextHolder.getContext().getAuthentication()).orElseThrow();
+    return Objects.equals(projectUser.userId(), userId);
   }
 
-  private Grid<ProjectAccessService.ProjectCollaborator> projectCollaboratorGrid() {
+  private void showControls(boolean isVisible) {
+    buttonBar.setVisible(isVisible);
+  }
 
-    Grid<ProjectAccessService.ProjectCollaborator> grid = new Grid<>();
-    Editor<ProjectCollaborator> editor = grid.getEditor();
-    Binder<ProjectCollaborator> binder = new Binder<>(ProjectCollaborator.class);
+  private void initProjectUserGrid() {
```

Comment:
This grid has a lot of columns.
Can you use the same Component to render a user in the first column only?
I feel like a lot of space is wasted with information about the users and the main focus of this page should be access management (which disappears to the right hand side)
![Screenshot 2024-07-22 at 10 35 36](https://github.com/user-attachments/assets/9b40ef44-90bd-495b-9165-b1df07ff734c)

Reply:
Good point i adapted the frontend component accordingly